extension/src/search.rs

```diff
@@ -180,13 +199,191 @@ pub fn init_table(
+            error!("failed to serialize metadata: {}", e);
+        }),
+    )?;
+
+    let search_columns = proj_params.columns
+        .iter()
+        .map(|col| format!("COALESCE({}, '')", col))
+        .collect::<Vec<String>>()
+        .join(" || ' ' || ");
+
+    let query = format!(
+        "SELECT {return_columns} FROM {schema}.{table}
+     WHERE to_tsvector('english', {search_columns})
+     @@ to_tsquery('english', '{query}')
+     LIMIT 10;",
+        schema = proj_params.schema,
+        table = proj_params.table,
+        return_columns = return_columns.join(", "),
+        search_columns = search_columns,  // Dynamically concatenate columns
+        query = query.split_whitespace().collect::<Vec<&str>>().join(" | ").replace("'", "''"),
+    );
+
+    Spi::connect(|client| {
+        let mut results: Vec<pgrx::JsonB> = Vec::new();
+        let tup_table = client.select(&query, None, None)?;
+
+        for row in tup_table {
+            let mut row_result = serde_json::json!({});
+
+            for col in &return_columns {
+                let col_value: Option<serde_json::Value> =
+                    row.get_by_name::<pgrx::JsonB, _>(col)
+                        .ok()
+                        .flatten()
+                        .map(|v| v.0) // JSONB values
+                        .or_else(|| row.get_by_name::<String, _>(col).ok().flatten().map(|v| serde_json::json!(v))) // TEXT values
+                        .or_else(|| row.get_by_name::<i32, _>(col).ok().flatten().map(|v| serde_json::json!(v))); // INT values
+
+                row_result[col] = col_value.unwrap_or(serde_json::Value::Null);
+            }
+
+            results.push(pgrx::JsonB(row_result));
+        }
+
+        Ok(results)
+    })
+
+}
+
+pub fn rrf_score(
+    rank: i32,
+    rrf_k: Option<i32>,
+) -> f32 {
+    // Default to 10 if not provided
+    let r = rrf_k.unwrap_or(10);
+    1.0 / (rank as f32 + r as f32)
+}
+
+pub struct SearchResult {
+    data: pgrx::JsonB,
+    rank: i32,
+}
+
+#[derive(Debug)]
+pub struct AllResults {
+    data: pgrx::JsonB,
+    full_text_rank: Option<i32>,
+    semantic_rank: Option<i32>,
+    rrf_score: f32,
+}
+
+// Read the search results and their ranks.
+// use the RRF formula to calculate the final score for each result.
+// Return the results in order of their final score.
+
+// Might have to have a flag somewhere to indicate if the search is full-text or semantic
 pub fn search(
     job_name: &str,
     query: &str,
     api_key: Option<String>,
     return_columns: Vec<String>,
     num_results: i32,
     where_clause: Option<String>,
+) -> Result<Vec<pgrx::JsonB>> {
+    // THIS IS JUST A DUMMY IMPLEMENTATION.
+    // IT HAS NOT BEEN TESTED
+
+    let full_text_results = full_text_search(job_name, query, return_columns.clone())?;
+    // let semantic_results = semantic_search(job_name, query, api_key, return_columns, num_results, where_clause)?;
```

Comment:
would be useful to be executing this. I can help troubleshoot it if there are issues.